uniswap-v2/src/token_retriever.rs

```diff
@@ -0,0 +1,43 @@
+use substreams::scalar::BigInt;
+use substreams::Hex;
+
+use crate::abi::erc20;
+use crate::pb::dex_amm::v1::Token;
+
+pub(crate) struct TokenRetriever {
+    token_address: Vec<u8>,
+}
+
+impl TokenRetriever {
+    pub(crate) fn new(token_address: Vec<u8>) -> Self {
+        TokenRetriever { token_address }
+    }
+
+    pub(crate) fn get_name(&self) -> String {
+        erc20::functions::Name {}
+            .call(self.token_address.to_vec())
+            .unwrap_or(String::new())
+    }
+
+    pub(crate) fn get_symbol(&self) -> String {
+        erc20::functions::Symbol {}
+            .call(self.token_address.to_vec())
+            .unwrap_or(String::new())
+    }
+
+    pub(crate) fn get_decimals(&self) -> BigInt {
+        erc20::functions::Decimals {}
+            .call(self.token_address.to_vec())
+            .unwrap_or(BigInt::from(18))
+    }
+
+    pub(crate) fn to_struct(&self) -> Token {
```

Comment:
I wonder if it makes more sense to add this as a trait to `Token` (e.g. `from_address`) instead of creating a new struct. @robinbernon what do you think?

Reply:
Yeh I guess it depends what things you would want from the token address. If you want all this information here to create the Token struct every time then yes - something like `impl From<Vec<u8>> for Token` could work well here. If there are cases where you might want just some of the data from the address then I'd say it could be good to keep things as is.